smart-contracts/contracts/utils/UnlockOwner.sol

```diff
@@ -0,0 +1,234 @@
+  }
+
+
+  /**
+   * helper to check if sender is mainnet DAO
+   * @notice 6648936 is the ID representing mainnet domain on Connext Bridge
+   */
+  function _isBridgedDAO(uint32 origin, address caller) internal view returns(bool) {
+    return msg.sender == bridgeAddress 
+      && origin == 6648936 
+      && caller == daoTimelockAddress;
+  }
+
+  /**
+   * helper used to get the proxyAdmin address from the Unlock contract
+   */
+  function _getProxyAdmin(address proxy) internal view returns (address) {
+    address proxyAdminAddress = IUnlock(proxy).getAdmin();
+    return proxyAdminAddress;
+  }
+
+  /**
+   * Internal helper to execute a call
+   * @param action the type of action to perform is decribed as a number. Currently, 
+   * two types of actions are available : `1` to pass directly the callData to the Unlock contract, 
+   * and `2` to trigger a contract upgrade through Unlock's ProxyAdmin
+   * @param value amount of (native) tokens to send with the contract call
+   * @param callData the data to pass in the call 
+   */
+  function _execAction(uint8 action, uint value, bytes memory callData) internal returns (address, bytes memory) {
+    
+    // check where to forward the call
+    address contractToCall;    
+    if(action == 1) { 
+      contractToCall = unlockAddress;
+    } else if (action == 2) { 
+      contractToCall = _getProxyAdmin(unlockAddress);
+    }
+
+    (bool success, bytes memory returnedData) = contractToCall.call{value: value}(callData);
+
+    // catch revert reason
+    if (success == false) {
+      assembly {
+        let ptr := mload(0x40)
+        let size := returndatasize()
+        returndatacopy(ptr, 0, size)
+        revert(ptr, size)
+      }
+    }
+
+    return (contractToCall, returnedData);
+  }
+
+  /**
+   * This function can be called by the multisig to remove itself, so only the DAO
+   * is left as sole owner of the Unlock contract
+   * @param _newMultisig the address of a multisig contract. If set to `address(0)` then
+   * the multisig will be entirely renounced from the current contract with no way to add it back
+   */
+  function changeMultisig(address _newMultisig) external {
+    if ( !_isMultisig()) {
+      revert Unauthorized(msg.sender);
+    }
+    multisigAddress = _newMultisig;
+  } 
+
+
+  /**
+   * Calling this function will execute directly a call to Unlock or a proxy upgrade
+   * @notice This function can only be called by the DAO on mainnet
+   * @param callData the encoded bytes should contains both the call data to be executed and 
+   *  the *action* code that follows `_execAction` pattern (see above).
+   * @return the address of the contract where code has been executed
+   */
+  function execDAO (bytes memory callData) external payable returns (address, bytes memory) {
+     if(!_isDAO()) {
+      revert Unauthorized(msg.sender);
+    }
+    // unpack calldata args
+    (uint8 action, bytes memory execCallData) = abi.decode(callData, (uint8, bytes));
```

Comment:
ok so based on today's conversation I think this is where we would want to "store" `execCallData` and add a time delay during which it can't be executed.
We would need to add a `executeQueued` call that anyone can trigger to execute a `execCallData` but only after time has passed. We should also add a replay-prevention mechanism :)